xlcalculator/ast_nodes.py

```diff
@@ -148,7 +152,9 @@ def eval(self, context):
             context.ranges[addr].value = data = func_xltypes.Array(range_cells)
             return data
 
-        return context.eval_cell(addr)
+        value = context.eval_cell(addr)
+        context.reset()
```

Comment:
It is unclear to me why you need `reset()`. Are we setting `context.sheet` anywhere else? To me it looks like a new attribute you are introducing.

Reply:
Oh you change it in evaluator.py.